src/Shouldly/ShouldlyExtensionMethods/ShouldBeNullExtensions.cs

```diff
@@ -9,38 +11,38 @@ namespace Shouldly
     public static class ShouldBeNullExtensions
     {
         [ContractAnnotation("actual:notnull => halt")]
-        public static void ShouldBeNull<T>(this T actual)
+        public static void ShouldBeNull<T>([MaybeNull] this T actual)
```

Comment:
If this method returns, the `actual` expression is definitely null. `[MaybeNull]` is the strongest the .NET annotations get. Threw it in for completeness.

```cs
var nonnullableThing = new object();
nonnullableThing.GetHashCode(); // No warning
nonnullableThing.ShouldBeNull();
nonnullableThing.GetHashCode(); // Warning that nonnullableThing may be null
```